internal/controller/postgrescluster/pgbackrest.go

```diff
@@ -1463,12 +1464,16 @@ func (r *Reconciler) reconcilePostgresClusterDataSource(ctx context.Context,
 		Name:      instanceName,
 		Namespace: cluster.GetNamespace(),
 	}}
+	clusterVolumes, err := r.observePersistentVolumeClaims(ctx, cluster)
```

Comment:
❓ I see `observePersistentVolumeClaims` is called before `reconcileDataSource`. Is it possible to pass those PVCs down here rather than make this API call?